src/util/units.ts

```diff
@@ -36,15 +76,16 @@ function convertUnit(fromVal, fromUnit, toUnit, adjustPrecision = true) {
   return adjustPrecision ? decimalAdjust('round', result.toVal, -8) : result.toVal;
 }
 
-function normalizeUnitsWhenPossible(val1, unit1, val2, unit2) {
+export function normalizeUnitsWhenPossible(val1: any, unit1: any, val2: any, unit2: any) {
   // If both units are CQL date units, return CQL date units
   const useCQLDateUnits =
-    CQL_TO_UCUM_DATE_UNITS[unit1] != null && CQL_TO_UCUM_DATE_UNITS[unit2] != null;
-  const resultConverter = unit => {
+    CQL_TO_UCUM_DATE_UNITS[unit1 as CQLUnitKey] != null &&
+    CQL_TO_UCUM_DATE_UNITS[unit2 as CQLUnitKey] != null;
```

Comment:
You could have typed `unit1` and `unit2` as `CQLUnitKey` in the function signature.  I'm assuming the you didn't because it caused more churn that just casting them here? (And same for `convertTocQLDateUnit` below).

Reply:
ah yeah. I think I initially did it as any, then later added in the CQLUnitKey type when the compiler yelled at me for any, and did the inline cast rather than typing the args itself. I'll update

Reply:
Oh yeah.... I remember why I didn't do this (I think).

Even though we can safely type `unit1` and `unit2` as `CQLUnitKey` in the function definition, the unit tests get mad. They are using arguments that aren't valid `CQLUnitKey`s, and so the it won't compile (see screenshots). After I saw this I think I was just like alright well I'm not gonna change the content of the test right now, and maybe it can really be something other than a `CQLUnitKey` in practice. Let me know if you think action is needed

![image](https://user-images.githubusercontent.com/16297930/154152135-90e43f6d-a4ed-4917-af7c-1e35fdf90c5d.png)
![image](https://user-images.githubusercontent.com/16297930/154152162-f9354d55-1023-4bd9-9a0f-104e5ad6380b.png)

Reply:
Ah good call, idk why I never even thought to change the actual line itself rather than just type cast.. I like the latter.


What approach makes sense in the function below? I could try to use `in` again, but not entirely sure how we'd want to rework this function without using a cast still. Here's a half-baked approach????

``` TypeScript
export function convertToCQLDateUnit(unit: any) {
  if (unit in CQL_TO_UCUM_DATE_UNITS) {
    // it's already a CQL unit, so return it as-is, removing trailing 's' if necessary (e.g., years -> year)
    return unit.replace(/s$/, '');
  }
  return UCUM_TO_CQL_DATE_UNITS[unit as UCUMUnitKey];
}
```

Reply:
A longer but more correct approach would probably be:
```ts
export function convertToCQLDateUnit(unit: any) {
  let dateUnit;
  if (unit in CQL_TO_UCUM_DATE_UNITS) {
    // it's already a CQL unit, so return it as-is, removing trailing 's' if necessary (e.g., years -> year)
    dateUnit = unit.replace(/s$/, '');
  } else if (unit in UCUM_TO_CQL_DATE_UNITS) {
    dateUnit = UCUM_TO_CQL_DATE_UNITS[unit]; // <-- is it smart enought to allow this? If not, cast it.
  }
  return dateUnit;
}
That's also a bit more obvious about what happens if it's in neither (undefined).  And eliminates multiple exit paths in the function too.
```

Reply:
Thanks for the suggestion :) 

the cast was necessary, sadly, but I like this better

[ce8fc0d](https://github.com/cqframework/cql-execution/pull/260/commits/ce8fc0d7bd97f25dec557acc9276f14898005ea9)